File: src/themes/catalog/classic/matrixManifest.js
// Matrix — quiet terminal.
//
// Re-palette per the Matrix mockup (Orbits_theme_MATRIX). The previous
// incarnation had a falling-rain canvas; the mockup moved Matrix away
// from that and toward a calmer shell/terminal surface — square 0-radius
// corners throughout, 1px pastel-green borders (#7EE787), JetBrains Mono
// everywhere, `> ` prompt prefixes on inputs. UI differentiation lives in
// src/styles/theme-skins.css; MatrixBackground now only paints scanlines
// and a faint grain on a warm-black wash.

/** @type {import('../../types.js').ThemeManifest} */
const manifest = {
  id: 'classic-matrix',
  name: 'Matrix',
  subtitle: 'Тихий терминал',
  family: 'atmospheric',
  colorScheme: 'dark',

  tokens: {
    '--orb-bg-rgb':       '12 18 14',
    '--orb-surface-rgb':  '17 24 19',
    '--orb-border-rgb':   '27 37 32',
    '--orb-text-rgb':     '208 221 210',
    '--orb-muted-rgb':    '106 128 112',
    '--orb-accent-rgb':   '126 231 135',
    '--orb-accent2-rgb':  '106 211 115',
    '--orb-success-rgb':  '126 231 135',
    '--orb-danger-rgb':   '244 63 94'
  },

  shape: {
    radiusButton: '0px',
    radiusCard: '0px',
    radiusModal: '2px',
    shadowCard: 'none',
    blurSurface: '0px'
  },

  typography: {
    fontHeading: "'JetBrains Mono', ui-monospace, monospace",
    fontBody: "'JetBrains Mono', ui-monospace, monospace",
    fontMono: "'JetBrains Mono', ui-monospace, monospace",
    letterSpacingHeading: '0',
    lineHeightBody: 1.6
  },

  performance: {
    // No canvas rain anymore — the background is pure CSS. Particle count
    // is irrelevant but we keep the hook in case we ever want to re-introduce
    // a subtle cursor-blink sprite later.
    minFPS: 24,
    degradeOnLowBattery: false,
    maxParticles: { desktop: 0, mobile: 0, lowEnd: 0 }
  },

  background: () => import('./MatrixBackground.jsx')
};

export default manifest;

File: lib/themes/catalog/classic/matrix_manifest.dart
// Matrix — quiet terminal. Port of
// `git_push/src/themes/catalog/classic/matrixManifest.js`.
//
// Square corners (radius 0) and JetBrains Mono everywhere are the defining
// features. The previous Matrix had falling-rain canvas; the 2026-04 mockup
// dropped that for a calmer terminal-shell aesthetic — `MatrixBackground`
// (when ported) only paints scanlines + grain on a warm-black wash.

import 'package:flutter/material.dart';

import '../../backgrounds/matrix_background.dart';
import '../../manifest.dart';

const ThemeManifest matrixManifest = ThemeManifest(
  id: 'classic-matrix',
  name: 'Matrix',
  subtitle: 'Тихий терминал',
  family: ThemeFamily.atmospheric,
  colorScheme: Brightness.dark,

  tokens: ThemeTokenColors(
    bg: Color.fromARGB(255, 12, 18, 14),
    surface: Color.fromARGB(255, 17, 24, 19),
    border: Color.fromARGB(255, 27, 37, 32),
    text: Color.fromARGB(255, 208, 221, 210),
    muted: Color.fromARGB(255, 106, 128, 112),
    accent: Color.fromARGB(255, 126, 231, 135),
    accent2: Color.fromARGB(255, 106, 211, 115),
    success: Color.fromARGB(255, 126, 231, 135),
    danger: Color.fromARGB(255, 244, 63, 94),
    // No blue ticks in terminal-land — fold delivery into the green accent.
    deliveryRead: Color.fromARGB(255, 126, 231, 135),
  ),

  shape: ThemeShape(
    radiusButton: 0,
    radiusCard: 0,
    radiusModal: 2,
    shadowCard: <BoxShadow>[], // explicit "no shadow" — matches JS 'none'
    blurSurface: 0,
  ),

  typography: ThemeTypography(
    fontHeading: 'JetBrainsMono',
    fontBody: 'JetBrainsMono',
    fontMono: 'JetBrainsMono',
    letterSpacingHeading: 0,
    lineHeightBody: 1.6,
  ),

  performance: ThemePerformance(
    // Background is pure CSS / no canvas in JS — particle counts are zero
    // and stay that way. Hook is kept in case we re-introduce a cursor-blink
    // sprite later.
    minFPS: 24,
    degradeOnLowBattery: false,
    maxParticlesDesktop: 0,
    maxParticlesMobile: 0,
    maxParticlesLowEnd: 0,
  ),

  background: _buildMatrix,
);

Widget _buildMatrix(BuildContext context) =>
    const MatrixBackground(manifest: matrixManifest);
